File: RoRServer/src/main/java/commands/game/CreateLocoCommand.java
package commands.game;

import java.util.UUID;
import commands.base.Command;
import models.game.Map;
import models.game.PlayerLoco;
import models.game.Square;
import models.session.GameSession;
import models.session.RoRSession;
/**
 * 
 * @author Isabel Rott, Michelle Le
 * Command fuer das Erstellen einer Lok
 */
public class CreateLocoCommand implements Command {

	private int xPos;
	private int yPos;
	private UUID playerId;
	protected RoRSession session;
	/**
	 * yPos, xPos müssen von den Bahnhöfen rausgelesen werden
	 * @param session
	 * @param messageInfo
	 */
	public CreateLocoCommand(RoRSession session, UUID playerId) {
		xPos = 8;
		yPos = 3;
		this.session = session;
		//xPos = messageInfo.getValueAsInt("xPos");
		//yPos = messageInfo.getValueAsInt("yPos");
		this.playerId = playerId;
	}

	@Override
	public void execute() {
		GameSession gameSession = (GameSession)session;
		
		Map map = gameSession.getMap();
		Square square = map.getSquare(xPos, yPos);
		
		//prüfen ob auf dem Square eine Rail liegt
		if(square.getPlaceableOnSquare() != null) {
			PlayerLoco loco = new PlayerLoco(session.getName(), square, playerId);
			((GameSession) session).addLoco(loco);
		}
	}
}

File: RoRClient/Commands/Game/Create/CreateLocoCommand.cs
using RoRClient.Commands.Base;
using RoRClient.Communication.DataTransferObject;
using RoRClient.Models.Game;
using RoRClient.Models.Session;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RoRClient.Commands.Game.Create
{
    class CreateLocoCommand : CommandBase
    {
        private Guid locoId;
        private Guid playerId;
        private int xPos;
        private int yPos;
        private Compass drivingDirection;

        public CreateLocoCommand(GameSession session, MessageInformation messageInformation) : base(session, messageInformation)
        {
            locoId = Guid.Parse(messageInformation.GetValueAsString("locoId"));
            xPos = messageInformation.GetValueAsInt("xPos");
            yPos = messageInformation.GetValueAsInt("yPos");
            drivingDirection = (Compass)Enum.Parse(typeof(Compass), messageInformation.GetValueAsString("drivingDirection"));
            playerId = Guid.Parse(messageInformation.GetValueAsString("playerId"));
        }

        public override void Execute()
        {
            Player player = session.GetPlayerById(playerId);
            Square square = session.Map.GetSquare(xPos, yPos);
            
            Loco loco = new Loco(locoId, drivingDirection, square);
            player.Loco = loco;

            ((GameSession)session).AddLoco(loco);
        }
    }
}
